pkg/metrics/probemetrics/probemetrics.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package probemetrics
+
+import (
+	"sync"
+
+	"github.com/cilium/tetragon/pkg/metrics"
+	"github.com/cilium/tetragon/pkg/metrics/consts"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type Key struct {
+	id     uint32
+	attach string
+}
+
+type Stat struct {
+	policy string
+	missed float64
+}
+
+var (
+	MissedProbes = metrics.NewBPFCounter(prometheus.NewDesc(
+		prometheus.BuildFQName(consts.MetricsNamespace, "", "missed_probes_total"),
+		"The total number of Tetragon probe missed per policy,probe.",
+		[]string{"policy", "attach"}, nil,
+	))
+
+	lock  sync.Mutex
+	stats = make(map[Key]*Stat)
```

Comment:
I don't think this map is needed at all. What we can do instead is put the loop from `getKprobeStats` directly in the `Collect` method. Then the metrics collector will read metrics directly from ebpf/link structs instead of this map.

Unless you need this map for something else than metrics?

Reply:
hum I'd need to move progs and the link_info logic into metrics code, which I don't like,
I'd like the kprobe specific code to stay in generickprobe.go ... it's also convenient
to manage the timer from there..